src/sec.rs

```diff
@@ -993,7 +997,55 @@ impl Sec {
     fn saga_finished(&mut self, done_data: SagaDoneData) {
         let saga_id = done_data.saga_id;
         let saga = self.sagas.remove(&saga_id).unwrap();
-        info!(&saga.log, "saga finished");
+
+        // Log every saga completion, regardless of what happened:
+        //
+        // * There will be exactly one log entry for every saga that finishes.
+        // * The message will always say "saga finished".
+        // * There is always a "result" field that's either "success" or
+        //   "failure".
+        //
+        // There are three cases:
+        //
+        // 1. Successful completion (no errors).  This will be an "info" level
+        //    message.
+        //
+        // 2. Clean failure (no errors during unwinding).  This will be a "warn"
+        //    level message and have "undo_result" = "success".
+        //
+        // 3. Stuck (an error was generated during unwinding).  This will be an
+        //    "error" level message and have "undo_result" = "failure".
+        //
+        // This should make it fairly straightforward to summarize the results
+        // programmatically.  The saga id and name are already in the saga.log
+        // object so that will also make it easy to summarize by saga name or to
+        // find the entry for a particular saga.
+        match &done_data.result.kind {
+            Ok(SagaResultOk { .. }) => {
+                info!(&saga.log, "saga finished"; "result" => "success");
+            }
+            Err(SagaResultErr {
+                error_node_name,
+                error_source,
+                undo_failure,
+            }) => {
+                let log = saga.log.new(o!(
+                    "result" => "failure",
+                    "action_error_node_name" =>
+                        format!("{:?}", error_node_name),
+                    "action_error_source" => format!("{:?}", error_source),
+                ));
+                if let Some(undo) = undo_failure {
+                    error!(&log, "saga finished";
+                        "undo_error_node_name" => ?undo.0,
+                        "undo_error" => ?undo.1,
```

Comment:
pedantic style nit, take it or leave it: i think restructuring the tuple in the `if let` pattern is a bit nicer to read:
```suggestion
                if let Some((node_name, error)) = undo_failure {
                    error!(&log, "saga finished";
                        "undo_error_node_name" => ?node_name,
                        "undo_error" => ?error,
```